hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/BlockManagerImpl.java

```diff
@@ -162,7 +157,13 @@ public long persistPutBlock(KeyValueContainer container,
           }
         }
 
-        db.getStore().putBlockByID(batch, incrementalEnabled, localID, data,
+        if (incrementalEnabled.get() && !VersionedDatanodeFeatures.isFinalized(
+            HDDSLayoutFeature.HBASE_SUPPORT)) {
```

Comment:
`finalizeBlock(..)` in `BlockManagerImpl` can also have this check and return error before finalize.
```
if (!VersionedDatanodeFeatures.isFinalized(
            HDDSLayoutFeature.HBASE_SUPPORT)) { }
```

Reply:
But since OM also checks upgrade finalize before and call will fail in OM itself, So in this case client will not call DN and I think this check is not required in DN.

Reply:
Agree. We should add the HDDSLayoutFeature.HBASE_SUPPORT check in finalizeBlock too. Although there is check in OM, the check in DN is also preferred. It's like the double insurance.